### PR TITLE
docs: fixed typos and improved spelling

### PR DESCRIPTION
**Pull request by Nicholas-SR**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Fixes typos and improves the spelling for the guide on built-in function development

### CODE REVIEW

**Comment by lumianph** [comment]
@Nicholas-SR  thanks for your contribution so much. I notice that "a SQL" and "an SQL " both are used in this doc. I believe either way is fine, but we just need to keep consistent throughout the doc.

**Review by lumianph — CHANGES_REQUESTED** [review]
@Nicholas-SR thanks for your contribution so much. I notice that "a SQL" and "an SQL " both are used in this doc. I believe either way is fine, but we just need to keep consistent throughout the doc.

**Comment by Nicholas-SR** [comment]
> @Nicholas-SR thanks for your contribution so much. I notice that "a SQL" and "an SQL " both are used in this doc. I believe either way is fine, but we just need to keep consistent throughout the doc.

@lumianph Good catch, I've fixed that now.

File: docs/en/built_in_function_develop_guide.md [hunk 1]
```diff
@@ -38,12 +38,12 @@ Developers can declare function in [hybridse/src/udf/udf.h](https://github.com/4
...
+- Function names are all lowercase, with underscores between words. Check [snake_case](https://en.wikipedia.org/wiki/Snake_case) for more details.
+- Function names should be clear and readable. Use names that describe the purpose or intent of the function.
 
 #### 2.1.3 C++ and SQL Data Type
 
-C++ built-in function can use limited data types, including BOOL, Numeric, String, Timestamp and Date.The correspondence between the SQL data type and the C++ data type is shown as followed:
+C++ built-in functions can use limited data types, including BOOL, Numeric, String, Timestamp and Date.The correspondence between the SQL data type and the C++ data type is shown as follows:
 
 - | SQL Type  | C++ Type           |
   | :-------- | :----------------- |
```
Comment (jingchen2222) [inline docs/en/built_in_function_develop_guide.md:49]: Could you please help use format int16_t to  `int16_t` in the table?

Comment (Nicholas-SR) [reply docs/en/built_in_function_develop_guide.md:49]: @jingchen2222 Good eye, I updated it.

**Comment by jingchen2222** [comment]
@Nicholas-SR Good Job! Thanks for your time and contribution.

**Review by jingchen2222 — APPROVED** [review]
LGTM

**Review by lumianph — APPROVED** [review]

**Comment by lumianph** [comment]
@Nicholas-SR  thanks for the contribution, if you are interested, you may join our slack workspace to interact with us for any contribution opportunities in the future.
https://join.slack.com/t/openmldb/shared_invite/zt-ozu3llie-K~hn9Ss1GZcFW2~K_L5sMg

**Comment by lumianph** [comment]
[community pr/issue tracking](https://trello.com/c/S4D8P57j/17-community-pr-issue-tracking)

**Pull request merged**